data-structures/lists/python/astronaut_list_2D.py

```diff
@@ -0,0 +1,284 @@
+    add_new_value_to_astronaut(all_astronauts, index, country)
+    
+    print(all_astronauts[index]) # Testing
+
+
+    ### UPDATE DAYS ###
+    print("\nUpdate the number of days an astronaut has been in space")
+    
+    # Update the number of days currently in space
+    index = int(input("Enter the index of the astronaut: "))
+    new_days = input("Enter the new number of days: ")
+    all_astronauts = update_days_in_space(all_astronauts, index, new_days)
+    #all_astronauts[6][DAYS_INDEX] = "62"
+
+    # Output days in space of astronaut at the given index
+    astronaut = all_astronauts[index]
+    print(astronaut)
+
+
+    ### LOOP 2D LIST ###
+    print("\nOuput each astronaut list and the values of each list")
+    
+    # Output each astronaut list and the values of each list
+    output_all_astronauts(astronauts)
+
+
+    ### LOOP CALCULATION ###
+    # Calculate the total number of days in space
+    total = calculate_total_days(all_astronauts)
+    print(f"Total number of days in space: {total}")
+
+
+    ### FIND NAME SUBSTRING (index) ###
+    print("\nSearch for an astronaut that contains a given name")
+    print("If more than one astronaut contains the substring, the first index will be returned")
+    
+    # Check whether one of the astronaut lists contains a name substring
+    name_to_find = input("Enter the name of an astronaut to find: ")
+    index = find_astronaut_return_index(all_astronauts, name_to_find)
+    
+    # Output the search results
+    if index == -1:
+        print("Astronaut not found in this data set")
+    else:
+        print(f"Astronaut found at index: {index}")
+        print(all_astronauts[index])
+
+
+    ### FIND NAME SUBSTRING (list) ###
+    print("\nSearch for any astronauts that contain a given name")
+    print("Return a list of all astronauts that contain the name substring")
+   
+    # Find v2 - search for "an"
+    found_astronauts = find_astronaut_return_list(all_astronauts, name_to_find)
+    
+    # Output the search results
+    if len(found_astronauts) == 0:
+        print(f"No astronauts found with a name containing {name_to_find}")
+    else:
+        print(f"Astronauts found:")
+        output_all_astronauts(found_astronauts)
+
+
+    ### DELETE BY NAME ###
+    print("\nDelete an astronaut list based on their name")
+    
+    # Delete the first astronaut with a given name
+    name_to_find = input("Enter the name of an astronaut to delete: ")
+    found_index = find_astronaut_return_index(all_astronauts, name_to_find)
+    all_astronauts = delete_astronaut(all_astronauts, found_index)
+
+
+    ### DELETE BY INDEX ###
+    print("\nDelete an astronaut list at a given index")
+    
+    # Delete an astronaut list at a given index
+    delete_at_index = int(input("Enter the index of the list to delete: "))
+    all_astronauts = delete_astronaut(all_astronauts, delete_at_index)
+
+    # Delete a value of an astronaut list at the given indices
```

Comment:
same comment as previosuly - this potentially messes up the constants - I am not sure why you would do this within the context of this application